CMake/SetFolder.cmake

```diff
@@ -0,0 +1,33 @@
+function(set_folder directory folder)
```

Comment:
It was unclear to me what this function does from the name alone. Maybe something like `dolphin_add_target_ide_folders_recursive`?